### ISSUE TITLE
Add new political reform

### ISSUE THREAD

**Issue by settintotrieste** [settintotrieste/Victoria-2-Cold-War-Enhancement-Mod-CWE#156]
Now that religious rights reform has been removed, I am thinking of adding a new reform to replace it. Perhaps public meetings?

**Comment by ARR8** [comment]
+1 to public meetings

**Comment by settintotrieste** [comment]
Great! We shall add public meetings

**Comment by settintotrieste** [comment]
Public meetings basic layout complete. The only thing left to do is to add the public meetings reforms to all the country history files.

The outline for the 1946 and 1992 startdate reforms will be no_meetings for governments with no elections, free_meetings for democracies and hms_governments and limited_meetings for colonial, mandate governments

**Comment by ARR8** [comment]
Will do, and also redo the reform graphics

**Comment by settintotrieste** [comment]
Thanks!

**Comment by settintotrieste** [comment]
Penal code reform

**Comment by DutchDylan** [comment]
If you're still taking suggestions for political reforms, perhaps a reform could be added concerning abortion which would influence population growth? 
Here's [two](https://puu.sh/wLbDX.jpg) [images](https://puu.sh/wLbWH.jpg) of how it would/could look ingame, these are from when I had implemented it for myself back when only NWO was around.

**Comment by ARR8** [comment]
@DutchDylan - That may be good as a social reform, so that the right will less readily support it. We'll keep it in mind, though population growth mechanics as a whole are left over from NWO and we haven't changed them much apart from basic rebalancing.

@settintotrieste - Something I saw recently that I think would make a very good political reform, and a better fit for the era than the penal code, is centralization. I'm not sure if you've seen it, but the new up-and-coming NMM-based mod, HFM, is planning on including a centralization reform in its next version. I think a similar implementation would fix a few problems we're having with our current decision-based implementation:

- First, every country, upon formation, for some reason starts with a 2% tax malus for "recently changing centralization" even if the country hadn't changed their political centralization for centuries.
  - This may help with some of the economic problems with the 1946 start people are having.
- It allows the other parties in the upper house to have a say on centralization issues, rather than just the ruling party. Decisions can't account for a party that would prefer to keep the status quo, even if it does not align with their policies.
- It allows a separate history for each country. I mentioned this in #227, and you were right in that adding an exception to every country is difficult as-is, but it would be relatively simple by a reform.
  - This is especially good for another issue I've been thinking about, and that is tying centralization into events. For example, under the current system, communist dictatorships are always highly centralized, but why would Yugoslavia under the 1974 constitution be so? That was extremely decentralized, and even when it happens the centralization modifier doesn't change.
  - This also ties in to #123, as a possible upside to radical centralization could be the integration of certain vassals and SARs, and a downside to radical decentralization the creation of certain SARs - or an upside, if they succeed economically.
- Multiple levels and gradual changes in centralization would also be possible, and better allow political realities. This could also be interesting to tie into politics, as high centralization should probably have many pops outside of urban areas and belonging to minority groups calling to decentralize, something that can't be represented fully with only two levels for democracies.

**Comment by settintotrieste** [comment]
@ARR8 I've contemplated making centralisation and decentralisation a reform when I first implemented the feature. However I ultimately decided against doing so for reasons of convenience and game engine limitations.

1. It is incredibly tedious to add the right centralisation reform to all countries in the history file. Making the implementation of this highly time inefficient.

2. Making ideology support determine the level of support for greater centralisation is also unworkable as a parties with the same ideology often support different levels of centralisation across countries. Eg. A Conservative party in the uk supports centralisation but a conservative one in the us supports federalism which is a form of decentralisation. Making it a reform also makes it difficult to script level of centralism in special countries as in all dictatorships there is only 1 party in the upper house and if we make communists support decentralisation to simulate the Yugoslav scenario then all other communist nations would also select decentralisation. 

3. Furthermore in the event of a government change we need to create backend processes to shift the level of centralisation by events as it would not make sense for communist nations to hold onto a decentralised government for longer if the AI did not choose to reform centralisation immediately and decides to wait another 6 months for reforms to become avaliable again. This conveniently solved in the existing system by sidestepping the limitations of the reform system altogether. 

4. The ability to create exceptions for major nations like Yugoslavia and implement the integration of certain vassals can be done with the existing system as a scope does exist for checking for country modifiers. 

It is much easier and cleaner to use a decision based system and since it is already there I feel that there is no need to change it radically in such a way.

**Comment by ARR8** [comment]
For point one: there are relatively few nations at game start, and many of the generic ones can have starting centralization set by government form, like we did for public meetings.

For point 2: No reason to make all communists support centralization, Yugoslavia and other such nations can have the reform set by event and locked by a flag. As for conservatives, I think the thing they have in common in the example is that they are opposed to change in the status quo of centralization. It is true that liberals in different countries may support different directions for centralization, but I think the use of conditions and changing the order of reform can create a workable system.

For point 3: I don't think there's any reason to set centralization in the background, as communists who achieved power democratically will change centralization gradually, and those who did so by revolution can have the reform set in the revolution event, as many reforms currently are.

Point 4 is true, but that applies equally to both systems.

For point 5: I think we mean different things. Currently, there is no way for nationalists to achieve their ends via diplomacy. Puppets are frequently used when simulating autonomy, such as for Republica Srpska, the recent addition of Bougainville, Gaza, trust territories, etc. This isn't as important, though, and doesn't have to be implemented at all.

I'd also like to add that doing it this way would help alleviate decision and political modifier clutter slightly, and make thematic sense to make centralization less lightly changed, as well as allow the player to view centralization via the ledger.

**Comment by settintotrieste** [comment]
1. For the 1946 start date perhaps but there is still the 1992 start date that we have to contend with which has far more countries to set the correct centralisation reform for.

2. I must say that this defeats the purpose of this being set as a reform as reforms should not be arbitrarily blocked off by scripted historical events or situations for specific countries as it is meant to be universally avaliable for all countries. Country specific scripting is more for decisions and events.

3. This is true but I noticed that the AI can sometimes fail to select reforms even though they can and this might hinder the ability for them to get the optimal centralisation level effect for the government type.

It is good to have better UI however it we must be careful to avoid causing more gameplay oddities for purely aesthetic changes.

**Comment by ARR8** [comment]
1. That's true, but we can still set it based on other reforms/government in the meantime, like for public meetings.
2. I think you've misunderstood me - what I mean is, when an event happens, such as the 1974 Yugoslav constitution, taking the choice of writing the constitution will change the centralization reform and lock it, but for that nation only. It would still be available for every other country. Think along the lines of the slavery debate flag for the buildup to the American Civil War in vanilla.
3. I think this can be managed by having, e.g. in your example, communist pops support changing the centralization in a communist regime. Although I would reiterate that it makes sense for communists to "focus on one thing at a time" if they have power by democratic means. They don't have unlimited political capital, especially without absolute support, and it would make sense if they only managed to get around to centralization after, for example, cementing their hold on power by abolishing the vote. And for revolutions, it's a non-issue.

I understand about gameplay oddities, but allowing an immediate change in centralization across the spectrum, while it works, is thematically odd. Plus, people like meaningful reforms.

I do think there are more upsides than downsides to this, and, even then, the downsides can mostly be worked around.

**Comment by settintotrieste** [comment]
I have an idea that might solve the problems of both a decision and reform system. Perhaps we can instead have a dual system where the reforms will determine the overall larger picture of centralisation or decentralisation while the decisions handle the finer aspects of it. 

For example, there will be a reform with unitary state or regionalism or federation in that order and for each reform there will be different level of control via decisions, with totalitarism, centralisation and decentralisation. The reforms will set the framework of the government while the decisions set the actual ruling party's attitude towards governing the regions.

This allows for complex situations to be simulated such as the Yugoslav constitution which established a federation but was still centrally controlled by the federal communist party. This also allows for more dynamism as it allows upper house support participation and using party participation.

**Comment by ARR8** [comment]
Oof - having a hard time wrapping my head around that. So effectively there would be six centralization levels? Seems like a lot...

From my understanding, while Yugoslavia was controlled only by the communist party, it had devolved many powers and different branches of the Party had control in the autonomous areas, making it effectively a classic federation under only nominal rule from Belgrade.

So while the party would have preferred totalitarianism, the constitution was written in such a way that, even if the biggest centralizing hardliner achieved power, the autonomous state would have been able to veto it - so the ruling party's attitude didn't have any real affect, unless they would have been able to change the reform, which, in this case, they could not have due to veto powers.

Of course, that's just Yugoslavia, and, in many other countries, any such reform if desired would still have needed plain-old political capital, or upper house composition.

Maybe changing the reforms could simply be blocked if the ruling party ideology isn't conducive to it, even if the whole upper house supports it? I'm not sure of an example in real life, but that would definitely account for the ruling party.

**Comment by settintotrieste** [comment]
There are 2 parallel levels of centralisation, within the party and in the overall government. 

Yugoslavia isn't such a good example. The USSR would be a better one where it was structured as a federation in the constitution but due to the ruling party practicing centralism with all local parties having to follow the decrees of the central Committe, it was effectively governed as a unitary state.

The redrafting of the consitution to alter the legal separation of powers between regions will still require upper house support but the party can side step it with the use of party whips in the local governments. This nuance can be simulated via this system.

**Comment by ARR8** [comment]
I would argue that that is a unitary state. The systems in-game show status quo - regardless of what's written in the constitution, the party with 100% control over the appartus of government used their clout to centralize, effectively a reform. Reforms in Vic2 are rather abstract, with not all of them having to be constitutional changes. To use an HPM-era example, enacting "freedom of womb" for slavery in the UK did not actually change any laws - rather, the state merely paid slaveowners for their slaves to free them, an example of reform outside of legislation.

To me, this still seems like a reform more than a decision, especially as the transition from revolutionary Russia to the USSR under Lenin involved many such examples of strengthening the party despite the ideals and constitutional protections of the Mensheviks, a gradual process through party policy written as state policy.

**Comment by settintotrieste** [comment]
However I would prefer to keep political reforms defined as constitutional changes as virtually all of them in this mod are constitutional ones. Eg. The right to have public meetings, censorship laws, voting rights, upper house composition rules, conscription, minorities' right to vote. The only exception that I can think of is civil service.

**Comment by ARR8** [comment]
I think that's drawing a line that might not be so clear. For example, immigration changes weren't part of the US constitution, but they were part of the PRC constitution. Most of the reforms are a combination of constitutional and legislative changes, and if that dichotomy doesn't exist in vanilla I wouldn't see a reason to set it in the mod. Conscription is also not a constitutional law for many countries, and there are things that are in the constitution, such as freedom of religion, that are represented as purely legislative in the mod currently, since the religion is only a de facto measure of the direction of pop conversion - a nominally secular state could still fund religious institutions which work outside of the bounds of the government. If all political reforms were constitutional changes, a lot of things would have to change in the mod. For example, the US can't make any constitutional changes based just on the upper house composition - and what about states with no constitution, like Israel?

**Comment by settintotrieste** [comment]
Constitution was not the best word for me to use, I think the law of the country would be a better term for what I am describing. Political reforms should be kept as legal changes to a country's governance structure.

**Comment by ARR8** [comment]
Isn't that exactly what centralization is? Changing to a federation/devolving authority to local governments/taking away powers from them all seems like governance structure to me.

And either way, shouldn't the definition of a political reform not be a change to governance structure, but a "political reform"? The name itself suggests the definition to me, and things like immigration and conscription aren't exactly governance but political issues which can be reformed or rolled back.

**Comment by settintotrieste** [comment]
But the way in which the bulk of the reforms are represented are in the form of statutory changes to the laws of the country pertaining to the governmental aspect of the country so that is what I believe we should try to follow.

Yes centralisation is definitely defined as a political reform. But the reason why I don't wish to make it a political reform in game is due to the reasons I've mentioned above unless it uses the hybrid system which can help to counter the flaws of the reform system.

**Comment by ARR8** [comment]
To be clear, the problem with setting it as a reform is this:
> I noticed that the AI can sometimes fail to select reforms even though they can and this might hinder the ability for them to get the optimal centralisation level effect for the government type.

?

**Comment by settintotrieste** [comment]
Yes and these other concerns as well mentioned above:

1. It is incredibly tedious to add the right centralisation reform to all countries in the history file. Making the implementation of this highly time inefficient.

2. Making ideology support determine the level of support for greater centralisation is also unworkable as a parties with the same ideology often support different levels of centralisation across countries. 

3. Furthermore in the event of a government change we need to create backend processes to shift the level of centralisation by events as it would not make sense for communist nations to hold onto a decentralised government for longer if the AI did not choose to reform centralisation immediately and decides to wait another 6 months for reforms to become avaliable again. This conveniently solved in the existing system by sidestepping this limitation of the reform system altogether.

4. The ability to create exceptions for major nations like Yugoslavia and implement the integration of certain vassals can be done with the existing system as a scope does exist for checking for country modifiers.

5. I must say that this defeats the purpose of this being set as a reform as reforms should not be arbitrarily blocked off by scripted historical events or situations for specific countries as it is meant to be universally avaliable for all countries. Country specific scripting is more for decisions and events.

**Comment by settintotrieste** [comment]
In addition, if the ruling party supports politics reform which is dependent on average_militancy in the country, all other political reforms become avaliable for the AI to select as well this makes it impossible to script individual AI centralisation level support.

**Comment by ARR8** [comment]
I believe I'd addressed most of these above.

1. I'm offering to do it. We can do it automatically based on things already set.
2. I think different countries support moving the reform in the same direction, which is the most important factor. Conservatives support leaving it as is, liberals support moving it down. That's the main advantage of the reform system. Status quo varies across countries, and conservatives universally support it, whatever it is.
3. That's unnecessary, as it's set automatically for a revolution, and the 6 month system is a feature as it represents gradual change in the event of overhauling an entire government. A newly-elected communist government in a democratic system has to reform things over time. For another example, look at Erdogan's Turkey today - he didn't immediately end democracy, but is instituting gradual "reform." I don't see this as a negative.
4. Yes, and it can be done with the reform system, too. This applies equally to both systems.
5. I don't see the difference between blocking off a reform and blocking off an event. Vanilla blocks off a reform for America, and several others in other, more generic situations. Ultimately, it can be done with one flag that is applied in a variety of events so as not to be noticeable.

The other issue, the one I quoted above, is one with no real easy solution. However, it can be chalked up to the inefficiency of government. The current instant, immediate change of centralization is also rather unrealistic.

Yes, it is true that it can't be individually scripted. However, conditions can be set to prevent it changing too easily. Either way, won't the AI prioritize reforms for which there are movements/that POPs actually want?

**Comment by settintotrieste** [comment]
1. That is great

2,3,7 have the same critical problem which is that I noticed that the AI does not seem to prioritise or sometimes even select any reforms despite being able to which severely hinders the ability for the AI to select the desired centralisation reform let alone social or other political reforms.

4. If can work with the existing system, there is no strong need in this area to remake the system to acheive the same functionality that the current system already provides.

5. As much as possible, I try to prevent this in this mod as my vision for reforms in the mod is one that all governments can select as long as they have enough UH support.

I agree with you on the immediate change, so the solution that I pursued to fix is it to implement the 2 year timer before a change can be implemented again. At least it will change within a short period of time as it we use the sreform system, the level may remain unchanged for decades as the AI only independently select new reforms once in a blue moon, based on what I have observed.
